### PR TITLE
Bug 1150880 - add regex handling for l10n repacks

### PR DESCRIPTION
**Pull request by camd**
This fixes [bug 1150880](https://bugzilla.mozilla.org/show_bug.cgi?id=1150880)

This adds handling for the following buildernames:

* Firefox mozilla-central linux nightly l10n x/3
* Firefox mozilla-central linux64 nightly l10n x/3
* Firefox mozilla-central macosx64 nightly l10n x/3
* Firefox mozilla-central win32 nightly l10n x/3
* Firefox mozilla-central win64 nightly l10n x/3

### CODE REVIEW

File: treeherder/etl/buildbot.py [hunk 1]
```diff
@@ -914,8 +914,10 @@
     "Unknown": "?",
 }
 
-# Match the job part number from buildernames such as "... mochitest-5"
-NUMBER_RE = re.compile(r".*-(\d+)$")
+# Match the job part number from buildernames such as "... mochitest-5" or
+# "l10n 2/3"
```
Comment (edmorley) [inline treeherder/etl/buildbot.py]: We don't enforce a max line length, so this doesn't need to be wrapped (80 chars is quite 1990s :-)).

File: treeherder/etl/buildbot.py [hunk 2]
```diff
@@ -914,8 +914,10 @@
...
+
+NUMBER_RE = re.compile(r"((?<=-)\d+|\d+(?=/\d+))(?:/\d+)?$")
```
Comment (edmorley) [inline treeherder/etl/buildbot.py]: This can be simplified:
```r"[ -](\d+)(?:/\d+)?$"```

**Comment by camd** [comment]
@edmorley I bow to your regex-fu, sir.  sfink, jgriffin, jgraham and I iterated on that for a bit in IRC trying to find one that would work.  :D

**Comment by edmorley** [comment]
It's merely a sad testament to how much of my life has been wasted playing "un-munge that string".
That and my regex may not even pass the tests ;-)

**Comment by camd** [comment]
haha..  Well it worked perfectly.  Looking at it, it makes perfect sense.  I tried things CLOSE to that, but not close enough, clearly...  :)

**Pull request merged**